synapse/storage/controllers/persist_events.py

```diff
@@ -934,6 +934,8 @@ async def _get_new_state_after_events(
             state_res_store=StateResolutionStore(self.main_store),
         )
 
+        full_state = await res.get_state(self._state_controller)
```

Comment:
this looks a bit lost. why do we do it here rather than just before (or even inline in) the `return`?